File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.py
num = int(input())
cnt = 1

a = num // 10
b = num % 10
c = a + b
if c > 10:
    c %= 10
    new = b * 10 + c       
elif c % 10 == 0:
    new = b * 10
else:
    new = b * 10 + c  
if num != new:
    while True:
        a = new // 10
        b = new % 10
        c = a + b
        if c > 10:
            c %= 10
            new = b * 10 + c       
        elif c % 10 == 0:
            new = b * 10
        else:
            new = b * 10 + c  
            
        cnt += 1

        if num == new:
            break

print(cnt)

File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
	static StringBuilder output = new StringBuilder();
	static StringTokenizer tokens;
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		int num = Integer.parseInt(input.readLine());
		int now = num;
		int next = 0;
		int cnt = 1;
		
		if (num < 10) {
			next = num * 11;
		} else {
			next = (num % 10) * 10 + (((num % 10) + (num / 10)) % 10);
		}
		
		now = next;
		
		while(now != num) {
			if(now < 10) {
				next = now * 11;
			} else {
				next = (now % 10) * 10 + (((now % 10) + (now / 10)) % 10);
			}
			
			cnt++;
			now = next;
		}
		
		System.out.println(cnt);
	}
}
